dashboard/apps/prototype/models.py

```diff
@@ -16,33 +19,116 @@ class Person(models.Model):
     is_contractor = models.BooleanField(default=False)
     job_title = models.CharField(max_length=128, null=True)
     is_current = models.BooleanField(default=True)
-    raw_data = JSONField()
+    raw_data = JSONField(null=True)
 
     def __str__(self):
         return self.name
 
     class Meta:
         verbose_name_plural = ugettext_lazy('People')
 
+    def rate_between(self, start_date, end_date):
+        """
+        average day rate in range
+        param: start_date: date object - beginning of time period for average
+        param: end_date: date object - end of time period for average
+        return: Decimal object - average day rate
+        """
+        rate_list = self.rates.between(start_date, end_date)
+        segments = []
+        for n, rate in enumerate(rate_list):
+            start = rate.start_date
+            try:
+                end = rate_list[n + 1].start_date - timedelta(days=1)
+            except IndexError:
+                end = end_date
+            segments.append(
+                (start, end, rate.rate_between(start, end))
+            )
+            rate.rate_between(start_date, end_date)
+
+        total_workdays = dec_workdays(start_date, end_date)
+
+        print(segments)
+
+        return average_rate_from_segments(segments, total_workdays)
+
+    def rate_on(self, on=None):
+        """
+        rate at time of date
+        param: on: date object - if no start or end then rate on specific date
+        return: Decimal object - rate on date
+        """
+        rate = self.rates.on(on=on)
+        return rate.rate_on(on) if rate else None
+
+
+class RatesManager(models.Manager):
+    use_for_related_fields = True
+
+    def on(self, on):
+        return self.get_queryset()\
+            .filter(start_date__lte=on)\
+            .order_by('-start_date')\
+            .first()
+
+    def between(self, start_date, end_date):
+        rates = list(self.get_queryset()
+            .filter(start_date__lte=end_date, start_date__gt=start_date)
+            .order_by('start_date'))
+        first = self.on(start_date)
+        if first:
```

Comment:
`first` could be already in the `rates` list, if the `start_date` of the rate is the `start_date`?